numpy/core/src/umath/scalarmath.c.src

```diff
@@ -1428,23 +1428,54 @@ static PyObject *
 }
 /**end repeat**/
 
+/**begin repeat
+ *
+ * #name = byte, ubyte, short, ushort, int, uint,
+ *             long, ulong, longlong, ulonglong,
+ *             half, float, double, longdouble,
+ *             cfloat, cdouble, clongdouble#
+ * #Name = Byte, UByte, Short, UShort, Int, UInt,
+ *             Long, ULong, LongLong, ULongLong,
+ *             Half, Float, Double, LongDouble,
+ *             CFloat, CDouble, CLongDouble#
+ * #cmplx = 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1#
+ * #to_ctype = , , , , , , , , , , npy_half_to_double, , , , , , #
+ * #which = float*17#
+ * #func = PyFloat_FromDouble*17#
+ */
+static NPY_INLINE PyObject *
+@name@_@which@(PyObject *obj)
+{
+#if @cmplx@
+    if (emit_complexwarning() < 0) {
+        return NULL;
+    }
+    return @func@(@to_ctype@(PyArrayScalar_VAL(obj, @Name@).real));
+#else
+    return @func@(@to_ctype@(PyArrayScalar_VAL(obj, @Name@)));
+#endif
+}
+/**end repeat**/
+
+
+#if !defined(NPY_PY3K)
+
```

Comment:
This splits the @name@_@which@ templates into two parts: one for python2 using `long` (below) and one for both 2 and 3 using `float` (above).

Reply:
Do we want to backport this, or shouldn't we just ignore python 2 for such a specific, rarely touched, code chunk?

Reply:
No need to backport. [NEP 14](https://numpy.org/neps/nep-0014-dropping-python2.7-proposal.html) is not clear on this, but I think we have not been actively removing python2 code from the master branch yet.